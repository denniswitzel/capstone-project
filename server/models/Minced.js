import mongoose from 'mongoose'

const MincedSchema = mongoose.Schema({
    category: String,
    title: String,
    ingredients: String,
    allergies: Array,
    image: String,
    rating: Array
})

export default mongoose.model('minced', MincedSchema, 'minced')