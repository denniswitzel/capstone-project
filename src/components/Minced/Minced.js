import React from 'react'
import minced from '../../mocks/minced.json'
import Headline from '../Headline/Headline'
import Product from '../Products/Product'


export default function Minced(){
    return(
        <>
        <Headline headline="Minced" />
        {minced.map((minced, index) => (
        <Product key={index} title={minced.title} image={minced.image} />
        ))}
        </>
    )
}