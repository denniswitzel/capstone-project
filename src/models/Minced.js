const mongoose = require('mongoose')

const MincedSchema = mongoose.Schema({
    category: String,
    title: String,
    ingredients: String,
    allergies: Array,
    image: String,
    rating: Array
})

module.exports = mongoose.model('minced', MincedSchema, 'minced')