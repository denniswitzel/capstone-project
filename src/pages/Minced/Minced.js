import React from 'react'
import minced from '../../mocks/minced.json'
import Product from '../../components/Products/Product'
import Headline from '../../components/Headline/Headline'
import PropTypes from 'prop-types'


Minced.propTypes = { 
    minced: PropTypes.arrayOf(PropTypes.object),
    title: PropTypes.string,
    image: PropTypes.string,
}

export default function Minced(){
    return(
        <>
        <Headline headline="Minced" />
        {minced.map((minced, index) => (
        <Product key={index} title={minced.title} image={minced.image} />
        ))}
        </>
    )
}
