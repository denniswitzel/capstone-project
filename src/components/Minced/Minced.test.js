import React from 'react'
import renderer from 'react-test-renderer'
import Minced from './Minced'
import { render } from '@testing-library/react'

describe('Minced', () => {
    const title = 'amidori - veggie minced'
    const image = 'https://raw.githubusercontent.com/denniswitzel/capstone-project/master/src/images/minced/Amidori-Veggie-Minced.png'

    it('display title and image', () => {
        const { getByText } = render(<Minced title={title} image={image} key={title}/>)
        expect(getByText(title)).toBeInTheDocument()
        expect(document.querySelector("img").getAttribute("src")).toBe(image)
    })
    it('renders correctly', () => {
        const tree = renderer.create(<Minced />)
        expect(tree).toMatchSnapshot()
      })
  })

