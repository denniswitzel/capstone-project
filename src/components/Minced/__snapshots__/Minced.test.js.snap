// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Minced renders correctly 1`] = `
Array [
  <section
    className="sc-AxjAm kQvphW"
  >
    <svg
      className="sc-AxiKw kteWBu"
      onClick={[Function]}
    >
      back-arrow.svg
    </svg>
    <h2
      className="sc-AxirZ DxuTA"
    >
      Minced
    </h2>
  </section>,
  <div
    className="sc-AxhCb RLztF"
  >
    <h3
      className="sc-AxhUy iFxeWf"
    >
      amidori - veggie minced
    </h3>
    <img
      alt="amidori - veggie minced"
      className="sc-AxgMl lgKYTG"
      src="https://raw.githubusercontent.com/denniswitzel/capstone-project/master/src/images/minced/Amidori-Veggie-Minced.png"
    />
  </div>,
  <div
    className="sc-AxhCb RLztF"
  >
    <h3
      className="sc-AxhUy iFxeWf"
    >
      Farm Way - Veganer Hack
    </h3>
    <img
      alt="Farm Way - Veganer Hack"
      className="sc-AxgMl lgKYTG"
      src="https://raw.githubusercontent.com/denniswitzel/capstone-project/master/src/images/minced/Farm-Way-Veganes-Hack.png"
    />
  </div>,
  <div
    className="sc-AxhCb RLztF"
  >
    <h3
      className="sc-AxhUy iFxeWf"
    >
      Garden Gourmet - Sensational Hack
    </h3>
    <img
      alt="Garden Gourmet - Sensational Hack"
      className="sc-AxgMl lgKYTG"
      src="https://raw.githubusercontent.com/denniswitzel/capstone-project/master/src/images/minced/Garden-Gourmet-Hack.png"
    />
  </div>,
  <div
    className="sc-AxhCb RLztF"
  >
    <h3
      className="sc-AxhUy iFxeWf"
    >
      Like meat - Like Hack
    </h3>
    <img
      alt="Like meat - Like Hack"
      className="sc-AxgMl lgKYTG"
      src="https://raw.githubusercontent.com/denniswitzel/capstone-project/master/src/images/minced/Like-meat-Hack.png"
    />
  </div>,
  <div
    className="sc-AxhCb RLztF"
  >
    <h3
      className="sc-AxhUy iFxeWf"
    >
      R端genwalder - Veganes M端hlen Hack
    </h3>
    <img
      alt="R端genwalder - Veganes M端hlen Hack"
      className="sc-AxgMl lgKYTG"
      src="https://raw.githubusercontent.com/denniswitzel/capstone-project/master/src/images/minced/Ruegenwalder-Muehlen-Hack.png"
    />
  </div>,
]
`;
